src/ORM/Rule/ExistsIn.php

```diff
@@ -140,6 +140,12 @@ public function __invoke(EntityInterface $entity, array $options): bool
             $primary,
             $entity->extract($fields)
         );
+        foreach ($conditions as $k => $v) {
+            if ($v === null) {
+                $conditions[$k . ' IS'] = $v;
```

Comment:
Using the same array seems wrong, better create a different one

Reply:
Or don't use array_combine and instead iterate the primary fields and build a condition array with a foreach.